go/vt/sqlparser/sql.y

```diff
@@ -2202,10 +2202,28 @@ length_opt:
     $$ = NewIntLiteral($2)
   }
 
+double_length_opt:
+  {
+    $$ = LengthScaleOption{}
+  }
+| '(' INTEGRAL ',' INTEGRAL ')'
+  {
+    $$ = LengthScaleOption{
+        Length: NewIntLiteral($2),
+        Scale: NewIntLiteral($4),
+    }
+  }
+
 float_length_opt:
   {
     $$ = LengthScaleOption{}
   }
+| '(' INTEGRAL ')'
+  {
+    $$ = LengthScaleOption{
+        Length: NewIntLiteral($2),
+    }
+  }
 | '(' INTEGRAL ',' INTEGRAL ')'
   {
     $$ = LengthScaleOption{
```

Comment:
nit: this can be slightly cleaned up to use the same newly created `double_length_opt`, like - 

```
float_length_opt:
double_length_opt
   {
     $$ = $1
   }
| '(' INTEGRAL ')'
   {
     $$ = LengthScaleOption{
         Length: NewIntLiteral($2),
     }
   }
```

Reply:
I don't know if this will cause a shift-reduce conflict, but I don't expect that to be the case.